src/watermarks.ml

```diff
@@ -163,10 +163,99 @@ let subst_file path ~map =
   | None -> ()
   | Some s -> Io.write_file path s
 
-let read_project_name () =
-  Dune_project.read_name (Path.in_source Dune_project.filename)
+(* Minimal API for dune-project files that makes as little assumption
+   about the contents as possible and keeps enough info for editing
+   the file. *)
+module Dune_project = struct
+  type 'a simple_field =
+    { loc : Loc.t
+    ; loc_of_arg : Loc.t
+    ; arg : 'a
+    }
 
-let get_name ~files ?name () =
+  type t =
+    { contents : string
+    ; name : string simple_field option
+    ; version : string simple_field option
+    }
+
+  let file = Path.in_source Dune_project.filename
+
+  let load file =
+    let s = Io.read_file file in
+    let lb = Lexing.from_string s in
+    lb.lex_curr_p <-
```

Comment:
Yh, but I also need the contents as a string

Reply:
Ah, that's a shame. Maybe we should have a combinator for this use case as well. At least this would mean we'd have the `lex_curr_p` initialization in only 1 place.

Reply:
Yh, I tried to find a good place to put a function that initialises `lex_curr_p`, but couldn't decide on one